src/components/atom_preference_language_checkbox/index.tsx

```diff
@@ -0,0 +1,48 @@
+import { GlobalLanguageCode } from '@/global.interface'
+import { preferenceState } from '@/recoil/preferences/preference.state'
+import { Checkbox, FormControlLabel } from '@mui/material'
+import { FC, SyntheticEvent } from 'react'
+import { useRecoilCallback, useRecoilValue } from 'recoil'
+
+interface Props {
+  languageCode: GlobalLanguageCode
+}
+
+const PreferenceLanguageCheckbox: FC<Props> = ({ languageCode }) => {
+  const preference = useRecoilValue(preferenceState)
+  const onChange = useRecoilCallback(
+    ({ set, snapshot }) =>
+      async (e: SyntheticEvent<Element, Event>, checked: boolean) => {
+        const preference = await snapshot.getPromise(preferenceState)
+        if (!preference) return
+
+        const element = e.target as HTMLInputElement
+        const languageCode = element.value as GlobalLanguageCode
+        const nativeLanguagesSet = new Set([
+          ...preference.nativeLanguages,
+          languageCode,
+        ])
+
+        if (!checked) nativeLanguagesSet.delete(languageCode)
+        const nativeLanguages: GlobalLanguageCode[] =
+          Array.from(nativeLanguagesSet)
+
+        set(preferenceState, {
+          ...preference,
+          nativeLanguages,
+        })
+      },
+  )
+
+  return (
+    <FormControlLabel
+      checked={preference?.nativeLanguages.includes(languageCode)}
+      control={<Checkbox />}
+      label={languageCode}
```

Comment:
https://github.com/ajktown/wordnote/blob/53cafd82bf0602ba1afbdbd83f5816d4d877dfa6/src/global.constants.ts#L50-L57

Use this function

Reply:
Handled in https://github.com/ajktown/wordnote/pull/101/commits/0db4b298f4261d9449cc2ff0ec00b4cefd7baaa6